### PR TITLE
#137091003/send invites to 11 other people

### PR DESCRIPTION
**Pull request by andela-mabdussalam**
#### What does this PR do?
Allows users to be invite users.
#### Description of Task to be completed?
Users are able to invite up to 11 users
#### How should this be manually tested?
Test on heroku

### CODE REVIEW

File: app/controllers/api/mailer.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require('dotenv').config();
```
Comment (andela-mabdussalam) [inline app/controllers/api/mailer.js]: Repetitive code. Already exists in server.js

File: app/controllers/api/mailer.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    You have been invited by <a>${req.user.name}</a> 
+    to join a game in cards for humanity<br/>It is a 
+    gracious way to give back to the human race.<br/>
+    <h3>click on this link <a href="${link}">here</a>
+    <h3> to join the game now.<br/>`
+  };
+
+  transporter.sendMail(mailOptions, (error, info) => {
+    if (error) {
+      res.json({ status: 'error' });
```
Comment (andela-mabdussalam) [inline app/controllers/api/mailer.js]: Add descriptive error message.

Comment (andela-mabdussalam) [reply app/controllers/api/mailer.js]: Also use return to avoid an "else" statement

File: app/controllers/api/search.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const mongoose = require('mongoose');
+ 
+ const User = mongoose.model('User');
```
Comment (andela-mabdussalam) [inline app/controllers/api/search.js]: Ensure proper indentation

File: app/controllers/api/search.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ 
+ /**
+  * Gets all users from the database
+  */
+ exports.users = (req, res) => {
+   if (req.user && req.user._id) {
```
Comment (andela-mabdussalam) [inline app/controllers/api/search.js]: Remove logic from if statement.

File: app/controllers/api/search.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+     const query = req.params.email || '';
+     User.find({ email: { $regex: query } }).limit(8)
+       .exec((err, result) => {
+         if (err) {
+           return res.json(err);
+         } else if (!result) {
+           
+           return res.status(404).send('No data found');
```
Comment (andela-mabdussalam) [inline app/controllers/api/search.js]: Send consistent error messages.

File: config/env/all.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ const rootPath = path.normalize(__dirname + '/../..');
 
 module.exports = {
   root: rootPath,
-  port: process.env.PORT || 3000,
-  db: process.env.MONGOHQ_URL,
+  port: process.env.PORT || 5000,
+  db: process.env.MONGOHQ_URL || 'mongodb://localhost:27017/cfh',
```
Comment (andela-mabdussalam) [inline config/env/all.js]: Use environment variables only for port and data url

File: config/routes.js [hunk 7]
```diff
@@ -92,4 +92,12 @@ module.exports = function(app, passport, auth) {
 
   var authn = require('../app/controllers/userAuth');
```
Comment (andela-mabdussalam) [inline config/routes.js]: Change to userAuth and replace var with const.

File: public/js/controllers/game.js [hunk 8]
```diff
@@ -201,5 +213,61 @@ angular.module('mean.system')
...
+          $('#modalView').modal('show');
+        return;
     }
-  ]);
+    if($scope.inviteList.includes($scope.email)){
+                $('#modalView1').modal('show');
+        return;
+    }
+    
+      $http({
```
Comment (nyambati) [inline public/js/controllers/game.js]: Abstract this functionality to services. Controllers should not be used to make `http` requests rather only to bind data to the Scope.

File: public/js/controllers/game.js [hunk 9]
```diff
@@ -201,5 +213,61 @@ angular.module('mean.system')
...
+         $scope.message = 'Could not send invite';
+       });
+       $scope.emailAddress = '';
+    };
+
+  
+$scope.searchUsers = function () {
+  $cope.isMailSent = false;
+  $scope.hideDiv = true;
+    $http.get(`/api/search/users/${$scope.email}`)
```
Comment (nyambati) [inline public/js/controllers/game.js]: This part should be abstracted to Angular services as well.

File: public/js/directives.js [hunk 10]
```diff
@@ -81,7 +81,15 @@ angular.module('mean.directives', [])
       templateUrl: '/views/timer.html',
       link: function(scope, elem, attr) {}
     };
-  }).directive('landing', function() {
+  })
+  .directive('invitelist', function() {
```
Comment (nyambati) [inline public/js/directives.js]: Directives should be named in camelCase. A good name for this should have been `inviteList`. And can be used in the UI as `<invite-list></invite-list>
`

**Review by nyambati — CHANGES_REQUESTED** [review]
You codebase has some few things that need your attention. These include indentation, Angular best practices, Proper use of conditionals. I have made a couple of comments, make the relevant changes

**Review by nyambati — APPROVED** [review]

**Pull request merged**